### PR TITLE
Bug 1967208: Use semver to parse version for "Getting started" card

### PR DESCRIPTION
**Pull request by spadgett**
Using the semver library is more robust than manually parsing the
string. Also fix the typo the hook name `useOpenshiftVersion` ->
`useOpenShiftVersion`.

/assign @jerolimov

### CODE REVIEW

File: frontend/packages/dev-console/src/components/guided-tour/GuidedTourText.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import { useOpenshiftVersion } from '@console/shared/src/hooks/version';
+import { useOpenShiftVersion } from '@console/shared/src/hooks/version';
 import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
 import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
 import { ConsoleLinkModel } from '@console/internal/models';
 
 const DevPerspectiveTourText: React.FC = () => {
   const { t } = useTranslation();
-  const openshiftVersion = useOpenshiftVersion();
+  const openshiftVersion = useOpenShiftVersion();
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/guided-tour/GuidedTourText.tsx:11]: Can / Want you want use semver here as well? I think the string slice here is more problematic as the other one.

**Review by jerolimov — CHANGES_REQUESTED** [review]
@spadgett see comment above

**Comment by spadgett** [comment]
@jerolimov Thanks, updated.

File: frontend/packages/dev-console/src/components/guided-tour/GuidedTourText.tsx [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+  const fullVersion = useOpenShiftVersion();
+  const parsed = semver.parse(fullVersion);
+  // Show only major and minor version.
+  const version = parsed ? `${parsed.major}.${parsed.minor}` : '4.x';
   return (
     <>
       {t(
-        'devconsole~Get started with a tour of some of the key areas in OpenShift {{version}} Developer perspective that can help you complete workflows and be more productive.',
-        { version: openshiftVersion ? [openshiftVersion?.slice(0, 3), "'s"].join('') : '4.x' },
+        "devconsole~Get started with a tour of some of the key areas in OpenShift {{version}}'s Developer perspective that can help you complete workflows and be more productive.",
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/guided-tour/GuidedTourText.tsx:19]: Expect that this need to update the localization files with `yarn i18n`

**Comment by spadgett** [comment]
Updated for `yarn i18n`, thanks.

**Review by jerolimov — APPROVED** [review]
Tested it locally. Works fine
/lgtm

**Pull request merged**